### PR TITLE
Add count_text_tokens, and expose operations.

### PR DESCRIPTION
**Pull request by MarkDaoust**
## Description of the change

* Add count_text_tokens
* Expose the operations service in the public api.
 
## Motivation

These should have been in v1beta2.

## Type of change

**Feature**

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have added detailed comments to my code where applicable.
- [ ] I have verified that my change does not break existing code.
- [ ] My PR is based on the latest changes of the main branch (if unsure, please run `git pull --rebase upstream main`).
- [ ] I am familiar with the [Google Style Guide](https://google.github.io/styleguide/) for the language I have coded in.
- [ ] I have read through the [Contributing Guide](https://github.com/google/generative-ai-python/blob/main/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: google/generativeai/models.py [hunk 1]
```diff
@@ -114,6 +114,31 @@ def get_tuned_model(
     return model_types.decode_tuned_model(result)
 
 
+def get_base_model_name(
+    model: model_types.AnyModelNameOptions, client: glm.ModelServiceClient | None = None
+):
```
Comment (markmcd) [inline google/generativeai/models.py:119]:
```suggestion
) -> str:
```

**Review by markmcd — APPROVED** [review]

**Comment by MarkDaoust** [comment]
Merge https://github.com/google/generative-ai-python/pull/78 first.

File: google/generativeai/text.py [hunk 2]
```diff
@@ -215,6 +216,23 @@ def _generate_response(
     return Completion(_client=client, **response)
 
 
+def count_text_tokens(
+    model: model_types.AnyModelNameOptions,
+    prompt: str,
+    client: glm.TextServiceClient = None,
```
Comment (markmcd) [inline google/generativeai/text.py:222]:
```suggestion
    client: glm.TextServiceClient | None = None,
```

File: google/generativeai/text.py [hunk 3]
```diff
@@ -215,6 +216,23 @@ def _generate_response(
...
+    base_model = models.get_base_model_name(model)
+
+    if client is None:
+        client = get_default_text_client()
+
+    result = client.count_text_tokens(
+        glm.CountTextTokensRequest(model=base_model, prompt={"text": prompt})
+    )
+
+    return type(result).to_dict(result)
```
Comment (markmcd) [inline google/generativeai/text.py:233]: Can we get a TypedDict or something for this return type? Even a `dict[str...]` or something would do as an interim.

File: tests/test_text.py [hunk 4]
```diff
@@ -367,6 +381,69 @@ def test_candidate_citations(self):
...
+    )
+    def test_count_message_tokens(self, model):
+        self.responses["get_tuned_model"] = glm.TunedModel(
+            name="tunedModels/bipedal-pangolin-001", base_model="models/text-bison-001"
+        )
+        self.responses["count_text_tokens"] = glm.CountTextTokensResponse(token_count=7)
+
+        response = text_service.count_text_tokens(model, "Tell me a story about a magic backpack.")
+        self.assertEqual({"token_count": 7}, response)
+        if len(self.observed_requests) > 1:
```
Comment (markmcd) [inline tests/test_text.py:441]: Can we make the guard condition here something that makes it clear why it's optional? e.g. `if make_model_name(model).startswith("tunedModel/"): ...` or an extra param like `should_look_up_model=bool`

**Review by markmcd — DISMISSED** [review]
Yay text counting :)

File: tests/test_text.py (hunk 4 above)
Comment (MarkDaoust) [reply tests/test_text.py:441]: Done.

**Review by markmcd — APPROVED** [review]

**Pull request merged**